src/runtime/gc_conservative.go

```diff
@@ -240,7 +240,7 @@ func calculateHeapAddresses() {
 	totalSize := heapEnd - heapStart
 
 	// Allocate some memory to keep 2 bits of information about every block.
-	metadataSize := totalSize / (blocksPerStateByte * bytesPerBlock)
+	metadataSize := (totalSize + blocksPerStateByte*bytesPerBlock + bytesPerBlock - 1) / (blocksPerStateByte*bytesPerBlock + bytesPerBlock)
```

Comment:
I think this should be 
```
metadataSize := 1+(totalSize / ( 1+blocksPerStateByte*bytesPerBlock )
```
Yet a better version is to use extra check:
```
	metadataSize := (totalSize / (1 + blocksPerStateByte*bytesPerBlock))
	for (totalSize-metadataSize)/bytesPerBlock > (metadataSize * blocksPerStateByte) {
		metadataSize++
	}

```
check this test: https://go.dev/play/p/3LRcr_8wZ38

Also, not sure how important this could be, it would be good if metadata start at aligned position.

Reply:
There are no alignment requirements on the metadata.